src/lib/loaders/loaders_without_authentication/weaviate.ts

```diff
@@ -0,0 +1,24 @@
+import factories from "../api"
+
+interface GraphQLArgs {
+  query: string
+  variables?: any
+}
+
+export default (opts) => {
+  const {
+    weaviateLoaderWithoutAuthenticationFactory: weaviateLoader,
+  } = factories(opts)
+
+  return {
+    weaviateGraphqlLoader: ({ query, variables }: GraphQLArgs) => {
+      return weaviateLoader(
+        "/graphql",
```

Comment:
How do we feel about using the [RESTful api](https://weaviate.io/developers/weaviate/api/rest)? I believe if we use their `graphql` api we would need to use stitching, which is in the process of being removed from MP.

Reply:
I don't think we need stitching since this is essentially a REST-like call but using a query format, right? We can simply use the query like this, for example:

```
const DiscoveryArtworks: GraphQLFieldConfig<void, ResolverContext> = {
  type: artworkConnection.connectionType,
  args: pageable({
    userId: { type: GraphQLString },
  }),
  resolve: async (_root, args, { weaviateGraphqlLoader, artworksLoader }) => {
    const { userId } = args

    const query = gql`
      query discoveryArtworksQuery {
        {
          Get {
            DiscoveryArtworks(where: { userId: "${userId}" }) {
              internalID
            }
          }
        }
      }
    `

    const body = await weaviateGraphqlLoader({
      query,
    })()

    if (!body.data.Get.DiscoveryArtworks) return connectionFromArray([], args)

    const artworkIds = body.data.Get.DiscoveryArtworks.map(
      (node) => node.internalID
    )

    const artworks =
      artworkIds?.length > 0 ? await artworksLoader({ ids: artworkIds }) : []

    return connectionFromArray(artworks, args)
  },
}
```

If this is not the preferred way we can def move to REST endpoints.

Reply:
After chatting, we will stick with `graphql`. Turns out the RESTful api is primarily built for DB management, and [not querying](https://weaviate.io/developers/weaviate/concepts/interface#restful-api-and-graphql-api).